system/jlib/jthread.cpp

```diff
@@ -702,22 +702,44 @@ void CAsyncFor::For(unsigned num,unsigned maxatonce,bool abortFollowingException
                 return 0;
             }
         };
-        if (maxatonce==0)
+        if (maxatonce==0 || maxatonce > num)
             maxatonce = num;
-        for (i=0;(i<num)&&(i<maxatonce);i++)
-            ready.signal();
-        IArrayOf<Thread> started;
-        started.ensure(num);
-        for (i=0;i<num;i++) {
-            ready.wait();
-            if (abortFollowingException && e) break;
-            Owned<Thread> thread = new cdothread(this,shuffled?shuffler->lookup(i):i,ready,&errmutex,e);
-            thread->start();
-            started.append(*thread.getClear());
+        if (maxatonce < num || abortFollowingException || shuffled)
```

Comment:
abortFollowingException isn't really supported if maxatonce == num, so could use the new code for that case, and shuffled would also be a trivial change to support.

Reply:
I agree it is fairly meaningless, so could ignore it.